### ISSUE TITLE
Add cargo-hack to the github ci action

### ISSUE THREAD

**Issue by itsyaasir** [collinsmuriuki/mpesa-rust#111]
Add https://crates.io/crates/cargo-hack to the CI action to test the crate with different features and if they compile correctly

**Comment by cpwallah** [comment]
okay can you assign this to me @itsyaasir

**Comment by itsyaasir** [comment]
You can do it

**Comment by cpwallah** [comment]
high @itsyaasir  can you give me link from where i can download openssl of 0.10 version because i am not able to find it

**Comment by itsyaasir** [comment]
Hello, the openssl is being pulled directly from the crates.io, are you getting errors with this ?

**Comment by c12i** [comment]
I think it should be getting pulled, if the features using the dependency are enabled

**Comment by cpwallah** [comment]
> Hello, the openssl is being pulled directly from the crates.io, are you getting errors with this ?

yes i am getting error when i am doing cargo build saying that i have no environment of openssl so try to set up the environment but i could not able to find 0.10 version of openssl @itsyaasir
